server/src/test/java/org/elasticsearch/persistent/PersistentTasksExecutorIT.java

```diff
@@ -307,6 +297,65 @@ public void testCreatePersistentTaskWithDuplicateId() throws Exception {
         });
     }
 
+    public void testUnassignRunningPersistentTask() throws Exception {
+        PersistentTasksClusterService persistentTasksClusterService =
+            internalCluster().getInstance(PersistentTasksClusterService.class, internalCluster().getMasterName());
+        // Speed up rechecks to a rate that is quicker than what settings would allow
+        persistentTasksClusterService.setRecheckInterval(TimeValue.timeValueMillis(1));
```

Comment:
Is there a risk to set it to a such low value?

Reply:
There is already another test that sets 1ms for the recheck interval.  I agree 1ms would be a completely inappropriate setting for production, but that's why it's being set via a method that end users cannot call.  I think if a test fails because the interval is low then it will be exposing a bug that _could_ happen with a higher setting, just much less frequently.  During this test it's true that the master node will be doing a lot of work iterating through the persistent tasks list, but it won't be doing the other work that a production master node would be doing, so a modern CPU should be able to cope.